src/plugins/analysis/hash/code/hash.py

```diff
@@ -1,43 +1,134 @@
+        sha384: Optional[str] = Field(description="sha384 hash of the file's content", default=None)
+        sha512: Optional[str] = Field(description="sha512 hash of the file's content", default=None)
+        blake2b: Optional[str] = Field(description="blake2b hash of the file's content", default=None)
+        blake2s: Optional[str] = Field(description="blake2s hash of the file's content", default=None)
+        sha3_224: Optional[str] = Field(description="sha3_224 hash of the file's content", default=None)
+        sha3_256: Optional[str] = Field(description="sha3_256 hash of the file's content", default=None)
+        sha3_384: Optional[str] = Field(description="sha3_384 hash of the file's content", default=None)
+        sha3_512: Optional[str] = Field(description="sha3_512 hash of the file's content", default=None)
+
+        # from helperFunctions.hash
+        ssdeep: Optional[str] = Field(description="ssdeep hash of the file's content", default=None)
+        tlsh: Optional[str] = Field(description="tlsh hash of the file's content", default=None)
+        imphash: Optional[str] = Field(
+            description="import hash of the executable's imported functions (only for ELF files)",
+            default=None,
+        )
+
+    def __init__(self):
+        super().__init__(
+            metadata=self.MetaData(
+                name='file_hashes',
+                description='calculate different hash values of the file',
+                version=Version(1, 3, 0),
+                dependencies=['file_type'],
+                Schema=self.Schema,
+            ),
+        )
+        configured_hashes = getattr(config.backend.plugin.get(self.NAME, None), 'hashes', [])
+        self.hashes_to_create = set(configured_hashes).union({'sha256', 'md5'})
+
+    def analyze(self, file_handle: FileIO, virtual_file_path: str, analyses: dict) -> Schema:
+        del virtual_file_path
+        result = {}
+
+        file_handle.seek(0)
+        file_contents = file_handle.read()
+        for hash_ in self.hashes_to_create.intersection(algorithms_guaranteed):
+            result[hash_] = get_hash(hash_, file_contents)
+        result['ssdeep'] = get_ssdeep(file_contents)
+        result['imphash'] = get_imphash(file_handle, analyses.get('file_type'))
+        result['tlsh'] = get_tlsh(file_contents)
+
+        return self.Schema(**result)
 
 
-class AnalysisPlugin(AnalysisBasePlugin):
+def get_imphash(file: FileIO, type_analysis: BaseModel | None) -> str | None:
     """
-    This Plugin creates several hashes of the file
+    Generates and returns the md5 hash of the (sorted) imported functions of an ELF file represented by `file_object`.
+    Returns `None` if there are no imports or if an exception occurs.
     """
+    if type_analysis is not None and _is_elf_file(type_analysis):
+        try:
+            with suppress_stdout():
+                functions = [f.name for f in lief.ELF.parse(file).imported_functions]
```

Comment:
Why not use `lief.logging.disable()`?

Reply:
I removed the call and everything related (function, test)